### ISSUE TITLE
Allow specific log filenames to be supplied to `ekat::logging::Logger`

### ISSUE THREAD

**Issue by jeff-cohere** [E3SM-Project/EKAT#209]
**Is your feature request related to a problem? Please describe.**

Currently,  all of the constructors for EKAT's `Logger` class append the MPI rank and `_logfile.txt` to the name provided. E3SM passes EAM a log file name to use for logging messages, so it would be nice to have a mechanism for specifying exactly that log file.

**Describe the solution you'd like**
Can we add a constructor to the `Logger` class that was able to accept a given log file name? I don't know how all of this fits in with MPI ranks and such. I think we should draw @AaronDonahue into this conversation.

**Comment by bartgol** [comment]
I completely agree. Instead of adding `_logfile.txt` we could simply add `.log`, since that's a widely recognized extension for log files.

As for the MPI ranks, we can also provide an optional ctor arg, that tells whether we should add rank info in the log file.

**Comment by bartgol** [comment]
Uhm, I am looking at the impl, and noticed that "rank0-only" is an optiion settable _after_ construction. Considering that such action (namely, turning off logging for rank>0) cannot be undone, it could easily be done at construction time. Even better, we could encode it with an extra template argument (which, I think, was the original design).

**Comment by bartgol** [comment]
I think this has been done. In case the mpi policy is "root only", the input filename is used as is (possibly adding a suffix, which is also customizable). The case where all ranks have their log still adds ".$rank.$size" to the filename, but I think that's healthy.
